plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -412,4 +399,96 @@ public function get_name(): string {
+	 *
+	 * When posts is authoritative and sync is enabled, deleting a post also deletes COT data.
+	 *
+	 * @param int     $postid The post id.
+	 * @param WP_Post $post The deleted post.
+	 */
+	private function handle_deleted_post( $postid, $post ): void {
+		if ( 'shop_order' === $post->post_type && ! $this->custom_orders_table_is_authoritative() && $this->data_sync_is_enabled() ) {
+			$this->data_store->delete_order_data_from_custom_order_tables( $postid );
+		}
+	}
+
+	/**
+	 * Handle the 'woocommerce_update_order' action.
+	 *
+	 * When posts is authoritative and sync is enabled, updating a post triggers a corresponding change in the COT table.
+	 *
+	 * @param int $order_id The order id.
+	 */
+	private function handle_updated_order( $order_id ): void {
+		if ( ! $this->custom_orders_table_is_authoritative() && $this->data_sync_is_enabled() ) {
+			$this->posts_to_cot_migrator->migrate_orders( array( $order_id ) );
+		}
+	}
+
+	/**
+	 * Handle the 'woocommerce_feature_description_tip' filter.
+	 *
+	 * When the COT feature is enabled and there are orders pending sync (in either direction),
+	 * show a "you should ync before disabling" warning under the feature in the features page.
+	 * Skip this if the UI prevents changing the feature enable status.
+	 *
+	 * @param string $desc_tip The original description tip for the feature.
+	 * @param string $feature_id The feature id.
+	 * @param bool   $ui_disabled True if the UI doesn't allow to enable or disable the feature.
+	 * @return string The new description tip for the feature.
+	 */
+	private function handle_feature_description_tip( $desc_tip, $feature_id, $ui_disabled ): string {
+		if ( 'custom_order_tables' !== $feature_id || $ui_disabled ) {
+			return $desc_tip;
+		}
+
+		$features_controller = wc_get_container()->get( FeaturesController::class );
+		$feature_is_enabled  = $features_controller->feature_is_enabled( 'custom_order_tables' );
+		if ( ! $feature_is_enabled ) {
+			return $desc_tip;
+		}
+
+		$pending_sync_count = $this->get_current_orders_pending_sync_count();
+		if ( ! $pending_sync_count ) {
+			return $desc_tip;
+		}
+
+		if ( $this->custom_orders_table_is_authoritative() ) {
+			$extra_tip = sprintf(
+				_n(
+					"⚠ There's one order pending sync from the orders table to the posts table. The feature shouldn't be disabled until this order is synchronized.",
+					"⚠ There are %1\$d orders pending sync from the orders table to the posts table. The feature shouldn't be disabled until these orders are synchronized.",
+					$pending_sync_count,
+					'woocommerce'
+				),
+				$pending_sync_count
+			);
+		} else {
+			$extra_tip = sprintf(
+				_n(
+					"⚠ There's one order pending sync from the posts table to the orders table. The feature shouldn't be disabled until this order is synchronized.",
+					"⚠ There are %%1\$d orders pending sync from the posts table to the orders table. The feature shouldn't be disabled until these orders are synchronized.",
```

Comment:
probably it should be a single % sign

Reply:
It is not related to the PR, sorry for making it looks confusing.